ext/AdvancedPSLibtaskExt.jl

```diff
@@ -81,26 +109,23 @@ function AdvancedPS.fork(trace::LibtaskTrace, isref::Bool=false)
     AdvancedPS.update_rng!(newtrace)
     isref && AdvancedPS.delete_retained!(newtrace.model.f)
     isref && delete_seeds!(newtrace)
-
-    # add backward reference
-    AdvancedPS.addreference!(newtrace.model.ctask.task, newtrace)
+    AdvancedPS.addreference!(newtrace)
     return newtrace
 end
 
 # PG requires keeping all randomness for the reference particle
 # Create new task and copy randomness
 function AdvancedPS.forkr(trace::LibtaskTrace)
+    rng = get_rng(trace)
     newf = AdvancedPS.reset_model(trace.model.f)
-    Random123.set_counter!(trace.rng, 1)
+    Random123.set_counter!(rng, 1)
+    trace.rng = rng
```

Comment:
Do we need `trace.rng = rng` if the previous line `set_counter!(rng, 1)` is mutating? 

```suggestion
    Random123.set_counter!(rng, 1)
    trace.rng = rng
```

Reply:
Seems we don't indeed, done.